packages/mermaid/src/rendering-util/rendering-elements/shapes/stateStart.ts

```diff
@@ -1,41 +1,41 @@
 import rough from 'roughjs';
-import type { SVG } from '../../../diagram-api/types.js';
 import type { Node, ShapeRenderOptions } from '../../types.js';
 import intersect from '../intersect/index.js';
 import { solidStateFill } from './handDrawnShapeStyles.js';
 import { updateNodeBounds } from './util.js';
+import type { D3Selection } from '../../../types.js';
 
-export const stateStart = (
-  parent: SVG,
+export function stateStart<T extends SVGGraphicsElement>(
+  parent: D3Selection<T>,
   node: Node,
   { config: { themeVariables } }: ShapeRenderOptions
-) => {
+) {
   const { lineColor } = themeVariables;
 
   const shapeSvg = parent
     .insert('g')
     .attr('class', 'node default')
     .attr('id', node.domId || node.id);
 
-  let circle;
+  let circle: D3Selection<SVGCircleElement> | D3Selection<SVGGElement>;
   if (node.look === 'handDrawn') {
     // @ts-ignore TODO: Fix rough typings
     const rc = rough.svg(shapeSvg);
     const roughNode = rc.circle(0, 0, 14, solidStateFill(lineColor));
     circle = shapeSvg.insert(() => roughNode);
+    // center the circle around its coordinate
+    circle.attr('class', 'state-start').attr('r', 7).attr('width', 14).attr('height', 14);
   } else {
     circle = shapeSvg.insert('circle', ':first-child');
+    // center the circle around its coordinate
+    circle.attr('class', 'state-start').attr('r', 7).attr('width', 14).attr('height', 14);
   }
 
-  // center the circle around its coordinate
-  // @ts-ignore TODO: Fix typings
-  circle.attr('class', 'state-start').attr('r', 7).attr('width', 14).attr('height', 14);
-
```

Comment:
These reason why we have to make this change is because TypeScript isn't smart enough to understand:

```
let circle: D3Selection<SVGCircleElement> | D3Selection<SVGGElement>;
circle.attr('class', 'state-start');
```

However, if we move this code into the `if()` blocks, then TypeScript knows that in one block it's `D3Selection<SVGCircleElement>`, and the other block it's `D3Selection<SVGGElement>`, and so it's happy to call `.attr` on it.